tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -1172,14 +1221,17 @@ operation::ProgramWithCallbacks Matmul::create_program(
                 );
             }
             else if constexpr (std::is_same_v<ProgramConfigType, MatmulMultiCoreReuseMultiCastProgramConfig>) {
+                bool mcast_use_same_noc = _get_use_same_noc(input_tensor_a, input_tensor_b, this->compute_kernel_config);
```

Comment:
does this need to be here? Can it be inside `matmul_multi_core_reuse_mcast_2d_optimized`? We should try to reduce the number of unnecessary args in our APIs

Reply:
ok, I will change it